### PR TITLE
Add options support

### PR DESCRIPTION
**Pull request by zce**
Some time we need custom timeout and hosts, so...

example: 

```js
var isOnline = require('is-online');
isOnline({ timeout: 1000, hostnames: ['youtu.be'] }, function(err, online) {
	console.log(online);
	// => true
});
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
When/why do you need that? Elaborating would help me improve this module.

**Comment by zce** [comment]
I want to improve the response speed when the electron application starts.

```javascript
// ...
isOnline(function (err, online) {
  if (online) {
    // load remote service
    mainWindow.loadURL('https://example.local/online.html')
    return
  }
  // load local page 
  // must wait two seconds !!!
  mainWindow.loadURL(`file://${__dirname}/offline.html`)
})
// ...
```

then 

```javascript
// ...
isOnline({ timeout: 500, hostnames: 'example.local' }function (err, online) {
  if (online) {
    // load remote service
    mainWindow.loadURL('https://example.local/online.html')
    return
  }
  // load local page 
  // must wait two seconds !!!
  mainWindow.loadURL(`file://${__dirname}/offline.html`)
})
// ...
```

and if connected, udp response time within 100ms.

thanks!

**Comment by silverwind** [comment]
Are you testing connectivity to a non-public host? I think `is-online` isn't really suitable for this, as both the module and `is-reachable` are built to only communicate with the public internet.

I think you might be better off just using a simple `net.connect`:

  https://nodejs.org/api/net.html#net_net_connect_port_host_connectlistener

**Comment by zce** [comment]
I also check the public network connection status.

Of course I can use `net.connect` to testing connectivity to my service.

I just hope that the module can improve the response efficiency by controlling the timeout period.

Thanks.

**Comment by silverwind** [comment]
+1 to configurable timeout. The current 2s is basically estimation of the maximum time needed to reach across the internet.

File: index.js [hunk 1]
```diff
@@ -4,11 +4,20 @@ var onetime = require('onetime');
 var roots = require('root-hints')('A');
 var isReachable = require('is-reachable');
 var randomItem = require('random-item');
-var hostnames = require('./hostnames');
 
-var timeout = 2000;
+var def = {
+	hostnames: require('./hostnames'),
```
Comment (sindresorhus) [inline index.js]: `require` should be where it is.

File: readme.md [hunk 2]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
+});
+```
+
+Custom options
+
+```js
+var isOnline = require('is-online');
+isOnline({ timeout: 1000, hostnames: ['youtu.be'] }, function(err, online) {
+	console.log(online);
+	// => true
```
Comment (sindresorhus) [inline readme.md]: Don't need example for this. It's not a common use-case.

File: readme.md [hunk 3]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
 });
 ```
 
 
 ## Node API
 
-### isOnline(callback)
+### isOnline([options,]callback)
```
Comment (sindresorhus) [inline readme.md]:
```
### isOnline([options], callback)
```

File: readme.md [hunk 4]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
+
+#### options
+
+*Optional*
```
Comment (sindresorhus) [inline readme.md]: Not needed

File: readme.md [hunk 5]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
 
 ## Node API
 
-### isOnline(callback)
+### isOnline([options,]callback)
+
+#### options
+
+*Optional*  
+Type: `object`
```
Comment (sindresorhus) [inline readme.md]: Object

File: readme.md [hunk 6]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
+
+##### timeout
+
+Type: `number`
+
+Milliseconds to wait for a server to send response.
+
+##### hostnames
+
+Type: `string`, `array`
```
Comment (sindresorhus) [inline readme.md]: Array

File: readme.md [hunk 7]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
...
+
```
Comment (sindresorhus) [inline readme.md]: Should mention the default value.

File: test.js [hunk 8]
```diff
@@ -10,3 +10,14 @@ test(function (t) {
 		t.assert(online);
 	});
 });
+
+console.time('online')
```
Comment (sindresorhus) [inline test.js]: You forgot to remove this

File: index.js [hunk 9]
```diff
@@ -4,11 +4,20 @@ var onetime = require('onetime');
...
+	timeout: 2000
+};
+
+module.exports = function (options, cb) {
+	if (typeof options === 'function') {
+		cb = options;
+		options = {};
+	}
+	var hostnames = options.hostnames || def.hostnames;
+	var timeout = options.timeout || def.timeout;
```
Comment (sindresorhus) [inline index.js]: Use https://github.com/sindresorhus/object-assign

**Comment by sindresorhus** [comment]
Ugh... Why can modules never stay simple.

File: readme.md [hunk 10]
```diff
@@ -16,19 +16,48 @@ $ npm install --save is-online
 
 ## Usage
 
+The basic use
```
Comment (silverwind) [inline readme.md]: Reword to `Basic usage`

**Comment by silverwind** [comment]
Regarding `timeout`: There are actually two timeouts involved. One for the DNS lookup (exposed here), and one for the TCP connections to the hostnames (exposed in https://github.com/sindresorhus/is-port-reachable/blob/3b8ef924693a1b7227563951ac44c71acce2312f/index.js#L13).

So I think before this can land, we need to add a `timeout` option to https://github.com/sindresorhus/is-reachable, which we can then expose as a second timeout option.

**Comment by sindresorhus** [comment]
ping @zce

**Comment by zce** [comment]
@sindresorhus I tried our best to make everything correct. If there are any errors or omissions , please let me know and forgive us. Thanks so much!

File: test.js [hunk 11]
```diff
@@ -11,12 +11,12 @@ test(function (t) {
...
 });
 
-console.time('online')
+// console.time('online')
 test(function (t) {
 	t.plan(2);
 
 	isOnline({ timeout: 500 }, function (err, online) {
-		console.timeEnd('online');
+		// console.timeEnd('online');
```
Comment (sindresorhus) [inline test.js]: remove, not comment out

File: index.js [hunk 12]
```diff
@@ -4,19 +4,21 @@ var onetime = require('onetime');
...
+var objectAssign = require('object-assign');
+var hostnames = require('./hostnames');
 
-var def = {
-	hostnames: require('./hostnames'),
-	timeout: 2000
-};
+var timeout = 2000;
+
+var defaults = { hostnames: hostnames, timeout: timeout }
```
Comment (sindresorhus) [inline index.js]: just put it inline in the objectAssign call

File: readme.md [hunk 13]
```diff
@@ -51,13 +39,13 @@ Type: `object`
 
 Type: `number`
 
-Milliseconds to wait for a server to send response.
+Milliseconds to wait for a server to send response. (Default: 2000)
```
Comment (sindresorhus) [inline readme.md]: put the `Default` below `Type`.

File: readme.md [hunk 14]
```diff
@@ -51,13 +39,13 @@ Type: `object`
...
 
 ##### hostnames
 
 Type: `string`, `array`
 
-One or more hosts to check.
+One or more hosts to check. (Default: ['www.google.com', 'www.cloudflare.com', 'www.baidu.com', 'www.yandex.ru'])
```
Comment (sindresorhus) [inline readme.md]: put the `Default` below `Type`.

File: index.js [hunk 15]
```diff
@@ -4,11 +4,22 @@ var onetime = require('onetime');
 var roots = require('root-hints')('A');
 var isReachable = require('is-reachable');
 var randomItem = require('random-item');
+var objectAssign = require('object-assign');
 var hostnames = require('./hostnames');
 
 var timeout = 2000;
```
Comment (sindresorhus) [inline index.js]: Just put it inline

File: readme.md [hunk 16]
```diff
@@ -21,14 +21,35 @@ var isOnline = require('is-online');
 
 isOnline(function(err, online) {
 	console.log(online);
-	//=> true
+	// => true
```
Comment (sindresorhus) [inline readme.md]: Don't change

File: index.js [hunk 17]
```diff
@@ -4,11 +4,20 @@ var onetime = require('onetime');
...
 
-module.exports = function (cb) {
+module.exports = function (options, cb) {
+	if (typeof options === 'function') {
+		cb = options;
+		options = {};
+	}
+
+	// Default options
```
Comment (sindresorhus) [inline index.js]: Remove comment

File: index.js [hunk 18]
```diff
@@ -4,11 +4,20 @@ var onetime = require('onetime');
...
+	options = objectAssign({ hostnames: hostnames, timeout: timeout }, options)
```
Comment (sindresorhus) [inline index.js]: Semicolon

**Pull request merged**